smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/annotations/OnOverflow.java

```diff
@@ -35,10 +35,17 @@
     enum Strategy {
         /**
          * Buffers <strong>all</strong> values until the downstream consumes it.
-         * This creates an unbound buffer if not used in conjunction with {@link #bufferSize()} to limit the size of the
-         * buffer. If the buffer is full, an error is propagated.
+         * This creates a buffer with the size specified by {@link #bufferSize()} if present.
+         * Otherwise, the size will be the value of the config property
+         * <strong>mp.messaging.emitter.defult-buffer-size</strong>.
+         * If the buffer is full, an error is propagated.
          */
         BUFFER,
+        /**
+         * Buffers <strong>all</strong> values until the downstream consumes it.
+         * This creates an unbound buffer. If the buffer is full, an error is propagated.
```

Comment:
_If the buffer is full_... in this case if the buffer is full, you are running out of memory :-)

Reply:
done